File: www/cgi-bin/html.php
#!/usr/bin/php-cgi
<?php
// En-tête HTTP obligatoire pour CGI
header("Content-Type: text/html; charset=UTF-8");


// Contenu HTML
echo "<!DOCTYPE html>";
echo "<html lang='fr'>";
echo "<head>";
echo "<meta charset='UTF-8'>";
echo "<title>Test PHP CGI</title>";
echo "</head>";
echo "<body>";
echo "<h1>Bonjour bonjour depuis PHP CGI !</h1>";
echo "<p>Ceci est une page HTML générée par un script PHP.</p>";
echo "</body>";
echo "</html>";
?>

File: www/cgi-bin/html.py
#!/usr/bin/python3
import os
import sys

sys.stdout.write("Content-Type: text/html; charset=UTF-8\r\n");
sys.stdout.write("\r\n");

print("<!DOCTYPE html>")
print("<html lang='fr'>")
print("<head><meta charset='UTF-8'><title>Test CGI Python</title></head>")
print("<body>")
print("<h1>Bonjour depuis Python CGI !</h1>")


print("<h2>Variables d'environnement CGI</h2>")
print("<ul>")
for key in [
    "REQUEST_METHOD", 
    "SCRIPT_NAME", 
    "SCRIPT_FILENAME", 
    "PATH_INFO", 
    "QUERY_STRING", 
    "REMOTE_ADDR", 
    ]:
    value = os.environ.get(key, "")
    if value:
        print(f"<li>{key} = {value}</li>")
print("</ul>")

print("</body></html>")
